### PR TITLE
feat(index) Automatically create indexes if they do not exist

### LINKED ISSUE #28802: Automatically start indexing if there is not a valid index on startup
### User Story

I see this on the dev image and I am sure the Cloud ENG team deals with this every day when cloning environments - if you point a dotCMS instance at a valid database and asset store with no ES indexes, dotCMS will start and run, but you still have to manually login, go into the backend Maintenance page and hit the "reindex" button before you will get any data in your ES index.

Awhile ago, we added the feature that if dotCMS starts up and does not see a valid index, it will create new live and working indexes, which is nice I guess but there is no difference between an empty index and no index.  The problem but it does not start the actual process of indexing the content.  That has to be done via a button click. 

This change will automatically start the reindexing process if dotCMS starts up with 0 contents in its index.

### Acceptance Criteria

If you start a dotCMS instance against a loaded DB with assets and no index, it should start normally and you should see the content getting indexed in the background.

### PR DESCRIPTION
**Pull request by wezell**
[this commit checks to see if live/working indexes are available at startup and if they are not, will automatically try to create and reindex into them. This allows a dotCMS instance to be restored using the db + assets and then, when started, dotCMS will automatically start the reindex process

ref: #28802

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java [hunk 1]
```diff
@@ -132,27 +130,62 @@ public synchronized void getRidOfOldIndex() throws DotDataException {
...
+                        Logger.error(this.getClass(), "Error starting reindex process", e);
+                    }
+                });
+
+            }
+
+
         } catch (Exception e) {
-            Logger.fatal("ESUil.checkAndInitializeIndex", e.getMessage());
+            Logger.fatal("ESUil.checkAndInitializeIndex", e.getMessage(),e);
```
Comment (erickgonzalez) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java:188]: typo?

**Review by erickgonzalez — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java [hunk 2]
```diff
@@ -132,27 +129,62 @@ public synchronized void getRidOfOldIndex() throws DotDataException {
...
         IndiciesInfo info = APILocator.getIndiciesAPI().loadIndicies();
-        return info.getWorking() != null && info.getLive() != null;
+
+
+        boolean hasWorking  = Try.of(()->APILocator.getESIndexAPI().indexExists(info.getWorking()))
+                .getOrElse(false);
+        boolean hasLive  = Try.of(()->APILocator.getESIndexAPI().indexExists(info.getLive()))
+                .getOrElse(false);
+
+        if(!hasWorking)Logger.debug(this.getClass(), "-- WORKING INDEX DOES NOT EXIST");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java:149]: use curly brackets

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java [hunk 3]
```diff
@@ -132,27 +129,62 @@ public synchronized void getRidOfOldIndex() throws DotDataException {
...
      */
-    private synchronized boolean indexReady() throws DotDataException {
+    @VisibleForTesting
+    @CloseDBIfOpened
+    public synchronized boolean indexReady() throws DotDataException {
         IndiciesInfo info = APILocator.getIndiciesAPI().loadIndicies();
-        return info.getWorking() != null && info.getLive() != null;
+
+
+        boolean hasWorking  = Try.of(()->APILocator.getESIndexAPI().indexExists(info.getWorking()))
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java:144]: set to final

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESIndexAPI.java [hunk 4]
```diff
@@ -690,30 +591,17 @@ public synchronized CreateIndexResponse createIndex(final String indexName, Stri
...
-		}
-		if (!map.containsKey("index.mapping.nested_fields.limit")) {
-			map.put("index.mapping.nested_fields.limit", 10000);
-		}
-
-		map.put("index.query.default_field",
-				Config.getStringProperty("ES_INDEX_QUERY_DEFAULT_FIELD", "catchall"));
+		map.putIfAbsent("index.mapping.total_fields.limit", 10000);
+		map.putIfAbsent("index.mapping.nested_fields.limit", 10000);
+		map.putIfAbsent("index.query.default_field", Config.getStringProperty("ES_INDEX_QUERY_DEFAULT_FIELD", "catchall"));
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESIndexAPI.java:602]: lazy?

**Review by jdotcms — APPROVED** [review]
Leave some minor comments but approving good stuff

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java (hunk 3 above)
Comment (wezell) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java]: done

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java (hunk 2 above)
Comment (wezell) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java]: fixed

**Pull request merged**